File: Snippet/C#/hello.cs
// C#
// Compile on the command line as follows: csc.exe hello.cs

using System;
using System.Reflection;
using System.Runtime.InteropServices;

class Test {
static void Main() {
Type t = Type.GetTypeFromProgID("Say.Tools");
object oST = Activator.CreateInstance(t);
string sText = "Hello world";
object[] aParams = {sText};
t.InvokeMember("Say", BindingFlags.InvokeMethod, null, oST, aParams);
}
}

File: Snippets/Visual Basic .NET/hello.vb
' Visual Basic .NET
' Compile on the command line as follows: vbc.exe hello.vb

Option Explicit

Module Test
Sub Main()
Dim oST As Object = CreateObject("Say.Tools")
Dim sText As String = "Hello world"
oST.Say(sText)
oST = Nothing
End Sub
End Module
